controllers/seasons.controllers.ts

```diff
@@ -1,7 +1,116 @@
-// TODO: require the model
-// Season = require("../models/Season")
+import { NextFunction, Response, Request } from "express";
+import Season from "../models/Season";
 
-// TODO: add controller logic 
+function create(req: Request, res: Response, next: NextFunction) {
+  const season = new Season({
+    name: req.body.name,
+    startDate: req.body.startDate,
+    endDate: req.body.endDate, //Think this should be calculated when status is changed to close
+    payroll_timeframe: req.body.payroll_timeframe,
+    price: req.body.price,
+    // TODO: Add product_id, unit_id, currency_id
+  });
 
-// TODO: export module
-// module.exports = { controllerName1, controllerName2, etc};
\ No newline at end of file
+  season
+    .save()
+    .then((result) => {
+      res.status(201).json({
+        message: "Season created",
```

Comment:
we have to chat with the team about the response structure of the endpoints.
probably we should handle the messages in the frontend

Reply:
https://github.com/LangaraVoyagers/project-2-api/pull/8/commits/51e7d10d24ca4d3388f1b9878249b8e91e2efd4a